src/expr.jl

```diff
@@ -1,26 +1,13 @@
 # This file contains default definitions for TermInterface methods on Julia
 # Builtin Expr type.
 
-istree(x::Expr) = true
-exprhead(e::Expr) = e.head
+iscall(x::Expr) = x.head == :call
 
-operation(e::Expr) = expr_operation(e, Val{exprhead(e)}())
-arguments(e::Expr) = expr_arguments(e, Val{exprhead(e)}())
+head(e::Expr) = e.head
+children(e::Expr) = e.args
 
-# See https://docs.julialang.org/en/v1/devdocs/ast/
-expr_operation(e::Expr, ::Union{Val{:call},Val{:macrocall}}) = e.args[1]
-expr_operation(e::Expr, ::Union{Val{:ref}}) = getindex
-expr_operation(e::Expr, ::Val{T}) where {T} = T
+# ^ this will implicitly define operation and arguments
```

Comment:
I would absolutely not define `operation` and `arguments` implicitly on Expr, or either *enforce* the "operation as first child" style. It's only relevant to Expr-like ASTs and thus should be up to the provider. I believe the current definitions of `operation` and `arguments` should be moved here.

This way, implementers of TermInterface may have to define 4 methods (`head`, `children`, `operation`, `args`), but they are not invited towards the default "operation as first child" behavior, but instead towards a `SymbolicUtils.Term` style, for most symbolic applications that involve `iscall`, which should remain optional

Reply:
This way I can just special-case reconstruction of `Expr`s in Metatheory and we are going to support all possible cases we discussed

Reply:
Agreed. Let's define operation and arguments explicitly for each term type.